internal/rbd/encryption.go

```diff
@@ -367,6 +374,34 @@ func (ri *rbdImage) initKMS(ctx context.Context, volOptions, credentials map[str
 	return nil
 }
 
+func ParseCipherOptions(volOptions map[string]string) (*cryptsetup.EncryptionOptions, error) {
+	cipher, cipherOk := volOptions["encryptionCipher"]
+	keysize, keysizeOk := volOptions["encryptionKeySize"]
+	integrity, integrityOk := volOptions["integrityMode"]
+	if !cipherOk {
+		return nil, nil
+	}
+	opts := &cryptsetup.EncryptionOptions{}
+	if cipherOk {
+		if err := opts.SetCipher(cipher); err != nil {
+			return nil, fmt.Errorf("failed to set cipher: %w", err)
+		}
+	}
+	// when cipher is not set keysize is not used
+	if cipherOk && keysizeOk {
+		if err := opts.SetKeySize(keysize); err != nil {
+			return nil, fmt.Errorf("failed to set key size: %w", err)
+		}
+	}
```

Comment:
```suggestion
if keysize, ok := volOptions["encryptionKeySize"];ok{
if err := opts.SetKeySize(keysize); err != nil {
			return nil, fmt.Errorf("failed to set key size: %w", err)
     }
}
```

lets group it together and have local loop, same for integrity as well